### DISCUSSION TITLE
can't get import to work

### DISCUSSION THREAD

**Discussion by joshmackay** [jwagner/simplex-noise.js#69] [Q&A]
Hi, I am having trouble getting this package to work. I have made a basic project and I cannot get the import to work in my JS file.

This is my JS file which works fine in the browser until I add the import line at the top.

`import { createNoise2D } from "simplex-noise";

function setup() {
	createCanvas(400, 400);
	noLoop();
}

function draw() {
	background(220);
}`

I added the "type" : "module", line to my package.json and added type="module" to my html script tag for the JS file.  This is the error I am left with....

_**Failed to resolve module specifier "simplex-noise". Relative references must start with either "/", "./", or "../".**_

Any ideas? I'm completely lost

**Comment by redblobgames** [comment]
Are you running that JS file directly in the browser, or do you have some build step in between?

**Comment by jwagner** [comment]
I'm assuming you aren't using any bundler but just a straight html file. In that case using a CDN like this might be easier to get you started without too much fiddling with es modules.
```
<script type="module">
import { createNoise2D } from 'https://cdn.jsdelivr.net/npm/simplex-noise@4.0.1/+esm'
const noise = createNoise2D();
console.log(noise(1.1,2.2));
</script>
```

If you don't want to use a CDN, [Parcel](https://parceljs.org/) is nice for a little basic project.
